### PR TITLE
Clippy fixes v2

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
None 

Describe changes:
- rust: fix clippy warnings in tests

@jasonish why do I get these and CI is not red ?
Is this because the lint level to deny warnings does not apply to test code ?

#10113 rebased and more commits
Only 19 match_single_binding warnings remaining

### CODE REVIEW

File: rust/src/tftp/tftp.rs [hunk 1]
```diff
@@ -225,13 +225,10 @@ mod test {
...
-                assert_eq!(tx, txp);
-            }
-            None => {
-                assert!(true);
-            }
+        if let Some(txp) = parse_tftp_request(&READ_REQUEST[..]) {
+            assert_eq!(tx, txp);
+        } else {
+            panic!("Result should not be None");
         }
```
Comment (jasonish) [inline rust/src/tftp/tftp.rs:232]: In tests I prefer:
```
        let txp = parse_tftp_request(&READ_REQUEST[..]).unwrap();
        assert_eq!(tx, txp);
```

File: rust/src/detect/byte_math.rs [hunk 2]
```diff
@@ -623,51 +620,37 @@ mod tests {
...
-                assert_eq!(val, bmd);
-            }
-            Err(_) => {
-                assert!(false);
-            }
+        if let Ok((_, val)) = parse_bytemath("bytes 4, offset 3933, oper +, rvalue myrvalue, result foo, string hex") {
+            assert_eq!(val, bmd);
+        } else {
+            panic!("Result shoud be ok");
         }
```
Comment (jasonish) [inline rust/src/detect/byte_math.rs:645]: Again, could just unwrap.

File: rust/src/dcerpc/dcerpc_udp.rs [hunk 3]
```diff
@@ -410,11 +410,8 @@ mod tests {
...
         ];
 
-        match parser::parse_dcerpc_udp_header(request) {
-            Ok((_rem, _header)) => {
-                { assert!(false); }
-            }
-            _ => {}
+        if parser::parse_dcerpc_udp_header(request).is_ok() {
+            panic!("Result shoud not be ok");
         }
```
Comment (jasonish) [inline rust/src/dcerpc/dcerpc_udp.rs:415]:
```
assert!(parser::parser_dcerpc_udp_header(request).is_err());
```

**Review by jasonish — CHANGES_REQUESTED** [review]
Happy to see this ints being fixed up, but I think we should opt for asserts and even unwraps/expects before conditional panics.

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 17375

File: rust/src/tftp/tftp.rs (hunk 1 above)
Comment (catenacyber) [reply rust/src/tftp/tftp.rs:232]: nice

File: rust/src/detect/byte_math.rs (hunk 2 above)
Comment (catenacyber) [reply rust/src/detect/byte_math.rs:645]: indeed

File: rust/src/dcerpc/dcerpc_udp.rs (hunk 3 above)
Comment (catenacyber) [reply rust/src/dcerpc/dcerpc_udp.rs:415]: yes

**Comment by catenacyber** [comment]
Replaces by https://github.com/OISF/suricata/pull/10129, thanks Jason for the review

**Pull request closed without merging**